File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    table1 = []
    table2 = []
    
    # 지도1 만들기
    for num in arr1:
        convertValue = bin(num)[2:]
        table1.append("0" * (n - len(convertValue)) + convertValue)
        
    # 지도2 만들기
    for num in arr2:
        convertValue = bin(num)[2:]
        table2.append("0" * (n - len(convertValue)) + convertValue)
    
    for i in range(n):
        tmp = ""
        for j in range(n):
            if (table1[i][j] == "1" or table2[i][j] == "1"):
                tmp += "#"
            else:
                tmp += " "
        answer.append(tmp)
        
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const answer = [];
    arr1 = arr1.map((item) => "0".repeat(n - item.toString(2).length) + item.toString(2));
    arr2 = arr2.map((item) => "0".repeat(n - item.toString(2).length) + item.toString(2));
    
    for (let i = 0; i < n; i++) {
        let result = "";
        for (let j = 0; j < n; j++) {
            if (arr1[i][j] === "1" || arr2[i][j] === "1") {
                result += "#"
            } else {
                result += " "
            }
        }
        answer.push(result)
    }
    
    return answer;
}